scipy/sparse/csgraph/tests/test_disjoint_set.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+from numpy.testing import assert_array_equal
+from scipy.sparse.csgraph import DisjointSet
+
+
+def test_linear_union_sequence():
+    n = 10
+    dis = DisjointSet(n)
+    assert dis.nc == n
+
+    for i in range(n - 1):
+        assert dis.union(i, i + 1)
+        assert dis.nc == n - 1 - i
+
+    parents = [dis.find(i) for i in range(n)]
+    assert_array_equal(0, parents)
+    assert not dis.union(0, n - 1)
+
+
+def test_self_unions():
+    n = 10
+    dis = DisjointSet(n)
+
+    for i in range(n):
+        assert not dis.union(i, i)
+    assert dis.nc == n
+
+    parents = [dis.find(i) for i in range(n)]
+    assert_array_equal(np.arange(n), parents)
+
```

Comment:
how about a more complicated join where find needs to recurse more than once, something like

```
def test_full_binary_tree():
    n = 16
    dis = DisjointSet(n)
    for i in range(0, 16, 2):
        assert not dis.union(i, i+1)
    for i in range(0, 16, 4):
        assert not dis.union(i+1, i+3)
    assert not dis.union(3, 7)

    parents = [dis.find(i) for i in range(n)]
    assert_array_equal([0,0,0,0,0,0,0,0,8,8,8,8,12,12,12,12], parents)
```

Reply:
Thanks for the suggestion. I have added a test to this effect.